src/sqlancer/postgres/gen/PostgresMergeGenerator.java

```diff
@@ -0,0 +1,378 @@
+        errors.add("MERGE condition must be a boolean expression");
+        errors.add("MERGE action must be INSERT, UPDATE, or DELETE");
+        errors.add("MERGE cannot be used with views");
+        errors.add("MERGE cannot be used with temporary tables");
+        errors.add("MERGE requires appropriate privileges");
+        errors.add("MERGE statement is not supported in this context");
+        errors.add("MERGE cannot be used in a function");
+        errors.add("MERGE cannot be used in a trigger");
+        errors.add("MERGE cannot be used in a stored procedure");
+        errors.add("MERGE statement is not allowed in this transaction mode");
+        errors.add("invalid reference to FROM-clause entry for table");
+        errors.add("column does not exist");
+        errors.add("relation does not exist");
+        errors.add("permission denied");
+        errors.add("cannot insert into column");
+        errors.add("cannot update column");
+        errors.add("ERROR: MERGE command cannot affect row a second time");
+        errors.add("ERROR: no collation was derived for column \".*\" with collatable type text");
+
+        try {
+            StringBuilder sb = new StringBuilder();
+
+            sb.append("MERGE INTO ");
+            sb.append(targetTable.getName());
+            sb.append(" AS target");
+
+            PostgresTable sourceTable = getSourceTable(globalState, targetTable);
+            List<PostgresColumn> sourceColumns = null;
+            boolean useSubquery = false;
+            if (sourceTable != null) {
+                sb.append(" USING ");
+                sb.append(sourceTable.getName());
+                sb.append(" AS source");
+            } else {
+                useSubquery = true;
+                sourceColumns = targetTable.getRandomNonEmptyColumnSubset();
+                sb.append(" USING (");
+                sb.append(generateSourceSubquery(globalState, targetTable, sourceColumns));
+                sb.append(") AS source");
+            }
+
+            sb.append(" ON ");
+            if (useSubquery) {
+                sb.append(generateMatchingCondition(targetTable, sourceColumns));
+            } else {
+                sb.append(generateMatchingCondition(targetTable, sourceTable));
+            }
+
+            boolean hasMatchedClause = false;
+            boolean hasNotMatchedClause = false;
+
+            if (Randomly.getBoolean()) {
+                sb.append(" WHEN MATCHED THEN");
+                hasMatchedClause = true;
+                if (Randomly.getBoolean()) {
+                    sb.append(" UPDATE SET ");
+                    try {
+                        if (useSubquery) {
+                            sb.append(generateUpdateSet(targetTable, sourceColumns));
+                        } else {
+                            sb.append(generateUpdateSet(targetTable, sourceTable));
+                        }
+                    } catch (Exception e) {
```

Comment:
This looks a bit surprising. Why does throwing an exception affect the logic of what we generate here? Can we achieve the intended outcome without relying on exceptions?